### PR TITLE
Hyrax 5795 works view item action file set delete throws error

### LINKED ISSUE #5795: Works view page - Items Actions - Delete not working
### Descriptive summary

The Delete option on the drop-down menu for Items on work view page is not working. Selecting Delete in drop-down menu gives rise to ArgumentError in Hyrax::GenericWorks#show.

### Expected behavior

Selecting delete from Items Actions drop-down menu should remove the file from the Work and reload the Work view page.

### Actual behavior

Following error loads:

Screen Shot 2022-08-12 at 9 59 32 AM

### Steps to reproduce the behavior

1. Log in to user dashboard, click work name to display work view page.
2. In the section of "Items", find the dropdown menu "Select an action" and select Delete from the drop-down menu and click OK at prompt to confirm delete action.
3. You will see an error like ArgumentError in Hyrax::GenericWorks#show

### Related work

#5136 (epic with history of previous work and testing of Works)
#5794 Issue for Download option in Items Actions menu
#5575 Issue for Edit and Versions options in Items Actions menu

### PR DESCRIPTION
**Pull request by fritzfreiheit**
Fixes #5795 ; refs #5136 #5794 #5575

Deleting a file set from a work does not entirely unlink file set.

The Delete option on the drop-down menu for Items on a work view page throws an error. Selecting Delete in drop-down menu raises ano ArgumentError in Hyrax::GenericWorks#show

Screen Shot 2022-08-12 at 9 59 32 AM

Changes proposed in this pull request:
* Unlink representative_id and template_id from work in remove from file set transaction step.
* Catch argument error from missing representative id resource in work show presenter.

Guidance for testing, such as acceptance criteria or new user interface behaviors:
* Log in to user dashboard, click work name to display work view page.
* In the section of "Items", find the dropdown menu "Select an action" and select Delete from the drop-down menu and click OK at prompt to confirm delete action.
* You will see an error like ArgumentError in Hyrax::GenericWorks#show

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/presenters/hyrax/work_show_presenter.rb [hunk 1]
```diff
@@ -98,6 +98,9 @@ def representative_presenter
           result = member_presenters([representative_id]).first
           return nil if result.try(:id) == id
           result.try(:representative_presenter) || result
+        rescue ArgumentError # Raised by valkyrie if representative_id object is not indexed
```
Comment (no-reply) [inline app/presenters/hyrax/work_show_presenter.rb:101]: this seems like a heavy handed rescue. won't this catch and swallow any ArgumentError regardless of the cause?

i think more precise handling is needed.

Comment (dlpierce) [reply app/presenters/hyrax/work_show_presenter.rb:101]: Yes, but that's the type of error raised by [PcdmMemberPresenterFactory#member_presenters](https://github.com/samvera/hyrax/blob/main/app/presenters/hyrax/pcdm_member_presenter_factory.rb#L64). We could change that to be a more unique error class.

Comment (no-reply) [reply app/presenters/hyrax/work_show_presenter.rb:101]: could we rework `SolrDocumentBehavior` so L95 `return nil if representative_id.blank?` handles this case?

Comment (dlpierce) [reply app/presenters/hyrax/work_show_presenter.rb:101]: The case this change is trying to address is when representative_id contains the id of a deleted or not yet indexed object, so SolrDocument is going to have a representative_id it assumes exists. We are trying to cover both ends of the issue by cleaning up when removing a fileset and making the presenter side less fragile.

Comment (no-reply) [reply app/presenters/hyrax/work_show_presenter.rb]: i like the changes that have come through on this. thanks!

**Review by dlpierce — APPROVED** [review]

**Pull request merged**